src/rooms/IPreRoomCreateHandler.ts

```diff
@@ -2,7 +2,14 @@ import { IRead } from '../accessors/index';
 import { IRoom } from './IRoom';
 
 export interface IPreRoomCreateHandler {
-    preRoomCreate(room: IRoom): IRoom;
+    /**
+     * Checks whether the handler actually shall execute.
+     * This method can basically implement a filter.
+     *
+     * @param room
+     * @param read
+     */
+    checkPreRoomCreate(room: IRoom, read: IRead): boolean;
 
-    isRoomApplicable(message: IRoom, read: IRead): boolean;
+    preRoomCreate(room: IRoom): IRoom;
```

Comment:
You would be create, this was just a simple interface to show a co-worker that classes can implement several interfaces and then showing how things get compiled down.